.github/workflows/weekly-summaries.yml

```
@@ -39,111 +39,118 @@ jobs:
+          # Verify and log thread generation
+          if [ -f "data/weekly/threads/latest.txt" ]; then
+            echo "Weekly thread generated successfully"
+            echo "Thread content:"
+            cat "data/weekly/threads/latest.txt"
+          else
+            echo "Warning: Thread generation may have failed"
+          fi
+
+      - name: Archive thread artifacts
+        if: env.IS_FRIDAY == '5'
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)